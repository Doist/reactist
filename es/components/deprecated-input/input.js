import { objectSpread2 as _objectSpread2 } from '../../_virtual/_rollupPluginBabelHelpers.js'
import * as React from 'react'
import classNames from 'classnames'

/**
 * @deprecated
 */

const Input = /*#__PURE__*/ React.forwardRef(function Input(props, ref) {
    const className = classNames('reactist_input', props.className)
    return /*#__PURE__*/ React.createElement(
        'input',
        _objectSpread2(
            _objectSpread2({}, props),
            {},
            {
                className: className,
                ref: ref,
            },
        ),
    )
})
Input.displayName = 'Input'

export { Input }
//# sourceMappingURL=input.js.map
