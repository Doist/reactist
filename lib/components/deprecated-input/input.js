"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("../../_virtual/_rollupPluginBabelHelpers.js"),t=require("react"),r=require("classnames");function u(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}function n(e){if(e&&e.__esModule)return e;var t=Object.create(null);return e&&Object.keys(e).forEach((function(r){if("default"!==r){var u=Object.getOwnPropertyDescriptor(e,r);Object.defineProperty(t,r,u.get?u:{enumerable:!0,get:function(){return e[r]}})}})),t.default=e,t}var a=n(t),c=u(r);const l=a.forwardRef((function(t,r){const u=c.default("reactist_input",t.className);return a.createElement("input",e.objectSpread2(e.objectSpread2({},t),{},{className:u,ref:r}))}));l.displayName="Input",exports.Input=l;
//# sourceMappingURL=input.js.map
