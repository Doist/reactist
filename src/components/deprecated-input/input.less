@import '../styles/constants.less';

.reactist_input {
    .normal_text();
    box-sizing: border-box;
    width: 100%;
    display: block;
    outline: none;
    border: 1px solid @color_border_grey;
    border-radius: 3px;
    padding: 5px 10px;
    margin: 0;

    &:focus {
        border-color: @primary_color;
    }
    &:disabled {
        background-color: @color_grey_11;
    }
}
