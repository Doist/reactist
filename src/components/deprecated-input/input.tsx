import React from 'react'
import classNames from 'classnames'

import './input.less'

type InputProps = {
    /** Additional css class applied to the input. */
    className?: string
    ref?: React.Ref<HTMLInputElement>
}

type Props = InputProps & React.InputHTMLAttributes<HTMLInputElement>

const Input: React.FC<React.PropsWithRef<Props>> = React.forwardRef(
    (props: Props, ref: React.Ref<HTMLInputElement>) => {
        const className = classNames('reactist_input', props.className)
        return <input {...props} className={className} ref={ref} />
    },
)
Input.displayName = 'Input'

export { Input }
