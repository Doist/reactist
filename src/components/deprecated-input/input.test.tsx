import React from 'react'
import { shallow } from 'enzyme'

import { Input } from './input'

describe('Input', () => {
    it('renders without crashing', () => {
        const input = shallow(<Input />)
        expect(input).toMatchSnapshot()
    })

    it('adds arbitrary props to the underlying input element', () => {
        const onChangeSpy = jest.fn()
        const input = shallow(<Input onChange={onChangeSpy} />)

        input.simulate('change')
        expect(onChangeSpy).toHaveBeenCalled()
    })

    it('adds additional className when supplied', () => {
        const input = shallow(<Input className="very-complex classnames-are-added" />)
        expect(input.hasClass('reactist_input')).toBe(true)
        expect(input.hasClass('very-complex')).toBe(true)
        expect(input.hasClass('classnames-are-added')).toBe(true)
    })
})
