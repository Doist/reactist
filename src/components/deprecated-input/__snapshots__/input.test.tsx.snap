// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders without crashing 1`] = `
<input
  className="reactist_input"
/>
`;
